src/Core/Managed/Shared/Extensibility/TelemetryConfiguration.cs

```diff
@@ -173,7 +245,26 @@ internal set
         /// <summary>
         /// Gets or sets the telemetry channel.
         /// </summary>
-        public ITelemetryChannel TelemetryChannel { get; set; }
+        public ITelemetryChannel TelemetryChannel
+        {
+            get
+            {
+                return this.telemetryChannel;
+            }
+
+            set
+            {
+                ITelemetryChannel oldChannel = this.telemetryChannel;
+                this.telemetryChannel = value;
+
+                // If we have a previously assigned channel which is not the same one as the "new" value
+                // passed in then we need to dispose of the old channel to keep from leaking resources.
+                if (oldChannel != null && oldChannel != value)
+                {
+                    oldChannel.Dispose();
```

Comment:
This is problematic, I realize `TelemetryConfiguration.Dispose` was already disposing the `TelemetryChannel`, however it doesn't own these objects - it receives them through property and constructor injection, so it shouldn't be disposing them. Combined with the `ObjectDisposedException` now thrown by the `InMemoryChannel` this is a significant change in behavior, unexpected from a well-behaved .NET API.